sdk/client.py

```diff
@@ -137,9 +137,11 @@ def _get_jobs(self, batch_id: int, fetch_results: bool = False) -> List:
         # Job results are not included when fetching a list of jobs
         # Fetch them explicitly if wanted
         if fetch_results:
+            results = self._request(
+                "GET", f"{self.endpoints.core}/api/v1/batches/{batch_id}/result"
+            )["data"]
             for job_data in job_list:
-                if job_data.get("status") == "DONE":
-                    job_data["result"] = self._get_job(job_data["id"]).get("result")
+                job_data["result"] = results.get(job_data["id"], None)
```

Comment:
I guess you're making this change so that all results are fetched at once ? Do you agree that in theory what was done before would still work ?
If so, shouldn't we also change the way it's done in `create_batch` when we wait for the results ? (indeed, that function still gets job results by calling `_get_job` for each job). 
If we do that, we'll probably need to update the batch fixture with some results, so that the test passes.

Reply:
Yeah the goal is to make a single API call instead of one for every job. In the `create_batch`function we need to retrieve all the jobs data and not only the results so I think we can't really change the implementation with something like this.

Reply:
Ok!